### PR TITLE
Reverting to TableView but with SwiftUI compatibility

### PR DESCRIPTION
**Pull request by Velin92**
Also if we really want to test and play around in nightly with SwiftUI I have set up a feature flag that allows us to change the timeline to SwiftUI whenever we wish.

### CODE REVIEW

File: ElementX/Sources/Application/AppSettings.swift [hunk 1]
```diff
@@ -213,4 +214,7 @@ final class AppSettings {
     
     @UserPreference(key: UserDefaultsKeys.notificationSettingsEnabled, defaultValue: false, storageType: .userDefaults(store))
     var notificationSettingsEnabled
+
+    @UserPreference(key: UserDefaultsKeys.swiftUITimelineEnabled, defaultValue: false, storageType: .userDefaults(store))
```
Comment (pixlwave) [inline ElementX/Sources/Application/AppSettings.swift:218]: Maybe we shouldn't persist this one so we know bugs reported are for whatever is active?

File: ElementX/Sources/Screens/RoomScreen/View/RoomScreen.swift [hunk 2]
```diff
@@ -82,13 +82,48 @@ struct RoomScreen: View {
...
+    @ViewBuilder
+    private var timelineSwitch: some View {
+        if context.viewState.swiftUITimelineEnabled {
+            TimelineView(viewState: context.viewState.timelineViewState,
+                         scrollToBottomButtonVisible: $context.scrollToBottomButtonVisible)
+        } else {
+            UITimelineView()
+                .overlay(alignment: .bottomTrailing) {
+                    scrollToBottomButton
+                }
```
Comment (pixlwave) [inline ElementX/Sources/Screens/RoomScreen/View/RoomScreen.swift:105]: Double checking: Does this overlay in the right place when the keyboard is shown? I know the order matters between the `overlay` and the `safeAreaInset` but can't remember which way round it is meant to be.

File: ElementX/Sources/Screens/RoomScreen/View/TimelineView.swift [hunk 3]
```diff
@@ -164,14 +164,14 @@ struct TimelineView: View {
 
 // MARK: - Previews
 
-struct TimelineTableView_Previews: PreviewProvider {
+struct UITimelineTableView_Previews: PreviewProvider {
```
Comment (pixlwave) [inline ElementX/Sources/Screens/RoomScreen/View/TimelineView.swift:168]: I think this should be?
```suggestion
struct TimelineView_Previews: PreviewProvider {
```

File: ElementX/Sources/Screens/RoomScreen/View/RoomScreen.swift [hunk 4]
```diff
@@ -82,13 +82,50 @@ struct RoomScreen: View {
             }
     }
 
+    @ViewBuilder
```
Comment (pixlwave) [inline ElementX/Sources/Screens/RoomScreen/View/RoomScreen.swift:85]: Does this one need to be `@ViewBuilder`?

**Review by pixlwave — APPROVED** [review]
*waves bye bye to smoothness for now* 😢

File: ElementX/Sources/Application/AppSettings.swift (hunk 1 above)
Comment (Velin92) [reply ElementX/Sources/Application/AppSettings.swift:218]: oh yeah better to keep it volatile I guess

File: ElementX/Sources/Screens/RoomScreen/View/RoomScreen.swift (hunk 2 above)
Comment (Velin92) [reply ElementX/Sources/Screens/RoomScreen/View/RoomScreen.swift:105]: looks good on my side, also i checked the previous code, this is the exact position it was present

**Pull request merged**